# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxFont.cpp
# Commit: c8f5a8ec70c4
# Full Hash: c8f5a8ec70c4e4fd2aef1defe3d27fb6cd831d3b
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2020-03-06 09:47:39
# Description:
#   Bug 1620125 - Don't assume mVerticalMetrics has necessarily been initialized before DrawOneGlyph is called. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D65567
# ==============================================================================

diff -r 5b935c9b36ab -r c8f5a8ec70c4 gfx/thebes/gfxFont.cpp
--- a/gfx/thebes/gfxFont.cpp	Thu Mar 05 17:44:44 2020 +0000
+++ b/gfx/thebes/gfxFont.cpp	Thu Mar 05 18:07:17 2020 +0000
@@ -1875,8 +1875,7 @@
 // coordinates (devPt) here.
 template <gfxFont::FontComplexityT FC>
 void gfxFont::DrawOneGlyph(uint32_t aGlyphID, const gfx::Point& aPt,
-                           GlyphBufferAzure& aBuffer,
-                           bool* aEmittedGlyphs) const {
+                           GlyphBufferAzure& aBuffer, bool* aEmittedGlyphs) {
   const TextRunDrawParams& runParams(aBuffer.mRunParams);
 
   gfx::Point devPt(ToDeviceUnits(aPt.x, runParams.devPerApp),
@@ -1897,7 +1896,8 @@
       // origin for each glyph, not once for the whole run.
       aBuffer.Flush();
       matrixRestore.SetContext(runParams.context);
-      gfx::Point skewPt(devPt.x + mVerticalMetrics->emHeight / 2, devPt.y);
+      gfx::Point skewPt(
+          devPt.x + GetMetrics(nsFontMetrics::eVertical).emHeight / 2, devPt.y);
       gfx::Matrix mat =
           runParams.context->CurrentMatrix()
               .PreTranslate(skewPt)